loadout.go

```diff
@@ -0,0 +1,411 @@
+		Loadout: options.Loadout.ToPlayerLoadout(),
+	})
+
+	conn.SendPacket(nil)
+
+	return nil
+}
+
+func WaitForMatchStart(conn *rlbot.RLBotConnection) (*flat.GamePacketT, error) {
+	var gamePacket *flat.GamePacketT
+	var fieldInfo *flat.FieldInfoT
+	for fieldInfo == nil || gamePacket == nil {
+		packet, err := conn.RecvPacket()
+		if err != nil {
+			return nil, err
+		}
+
+		switch packet := packet.(type) {
+		case *flat.FieldInfoT:
+			fieldInfo = packet
+		case *flat.GamePacketT:
+			gamePacket = packet
+		}
+	}
+
+	// while the match isn't active
+	for !(gamePacket.MatchInfo.MatchPhase == flat.MatchPhaseActive || gamePacket.MatchInfo.MatchPhase == flat.MatchPhaseKickoff) {
+		packet, err := conn.RecvPacket()
+		if err != nil {
+			return nil, err
+		}
+
+		switch packet := packet.(type) {
+		case *flat.GamePacketT:
+			gamePacket = packet
+		}
+	}
+
+	return gamePacket, nil
+}
+
+func (a *App) StaticSetter() error {
+	conn, err := rlbot.Connect(a.rlbot_address)
+	if err != nil {
+		return err
+	}
+
+	conn.SendPacket(&flat.ConnectionSettingsT{
+		AgentId:              "",
+		WantsBallPredictions: false,
+		WantsComms:           false,
+		CloseBetweenMatches:  true,
+	})
+
+	conn.SendPacket(&flat.InitCompleteT{})
+
+	_, err = WaitForMatchStart(&conn)
+	if err != nil {
+		return err
+	}
+
+	gameState := flat.DesiredGameStateT{
+		CarStates: []*flat.DesiredCarStateT{
+			{
+				Physics: &flat.DesiredPhysicsT{
+					Location:        Vector3P(0, 0, 20),
+					Rotation:        RotatorP(0, 0, 0),
+					Velocity:        Vector3P(0, 0, 0),
+					AngularVelocity: Vector3P(0, 0, 0),
+				},
+			},
+		},
+	}
+
+	for {
+		packet, err := conn.RecvPacket()
+		if err != nil {
+			return err
+		}
+
+		switch packet.(type) {
+		case nil:
+			return nil
+		case *flat.GamePacketT:
+			conn.SendPacket(&gameState)
+		}
+	}
+}
+
+func (a *App) SetPreviewInput(controller flat.ControllerStateT) error {
+	conn, err := rlbot.Connect(a.rlbot_address)
+	if err != nil {
+		return err
+	}
+
+	conn.SendPacket(&flat.ConnectionSettingsT{
+		AgentId:              "",
+		WantsBallPredictions: false,
+		WantsComms:           false,
+		CloseBetweenMatches:  true,
+	})
+
+	conn.SendPacket(&flat.InitCompleteT{})
+
+	_, err = WaitForMatchStart(&conn)
+	if err != nil {
+		return err
+	}
+
+	for {
+		packet, err := conn.RecvPacket()
+		if err != nil {
+			return err
+		}
+
+		switch packet.(type) {
+		case nil:
+			return nil
+		case *flat.GamePacketT:
+			conn.SendPacket(&flat.PlayerInputT{
+				PlayerIndex:     0,
+				ControllerState: &controller,
+			})
+		}
+	}
+}
+
+func (a *App) SetShowcaseType(showcaseType string) error {
+	conn, err := rlbot.Connect(a.rlbot_address)
+	if err != nil {
+		return err
+	}
+
+	conn.SendPacket(&flat.ConnectionSettingsT{
+		AgentId:              "",
+		WantsBallPredictions: false,
+		WantsComms:           false,
+		CloseBetweenMatches:  false,
+	})
+
+	conn.SendPacket(&flat.InitCompleteT{})
+
+	gamePacket, err := WaitForMatchStart(&conn)
```

Comment:
this line doesn't properly wait for a new car to spawn when a new match isn't started but C&S is used instead.

Reply:
Maybe, for the SetPreviewInput function you could also set a bounding box and if the car leaves it, teleport it back to some starting location.

Reply:
Or you could change WaitForMatchStart to WaitForMatchStartAndCarOnCorrectTeam.

Reply:
I mean, we could also just have both cars in the game all the time. Why not preview both at the same time 🤷